File: hashing.go
package main

import (
	"crypto/ed25519"
	"crypto/rand"
	"fmt"
	"time"
)

func main() {
	// Generate a new public/private keypair
	_, privateKey, err := ed25519.GenerateKey(rand.Reader)
	if err != nil {
		fmt.Println("Error generating key:", err)
		return
	}

	// Message to sign
	message := []byte("Hello, World!")

	// Measure time to sign
	totalIterations := 100000
	var totalDuration time.Duration

	for i := 0; i < totalIterations; i++ {
		start := time.Now()
		_ = ed25519.Sign(privateKey, message)
		duration := time.Since(start)
		totalDuration += duration
	}

	avgDuration := totalDuration / time.Duration(totalIterations)

	fmt.Printf("Average hashing/signing time: %v\n", avgDuration)
	fmt.Printf("Total time for %d iterations: %v\n", totalIterations, totalDuration)
}

File: hashing.rs
use ed25519_dalek::{Keypair, Signer};
use rand::rngs::OsRng;
use std::time::Instant;

fn main() {
    // Create a new keypair using a cryptographic random number generator
    let mut csprng = OsRng;
    let keypair = Keypair::generate(&mut csprng);

    // Message to sign
    let message = b"Hello, World!";

    // Measure time to sign
    let total_iterations = 100000;
    let mut durations = vec![];

    for _ in 0..total_iterations {
        let start = Instant::now();
        let _signature = keypair.sign(message);
        let duration = start.elapsed();
        durations.push(duration);
    }

    // Calculate total time and average time
    let total_time: std::time::Duration = durations.iter().sum();
    let avg_time = total_time / total_iterations as u32;

    println!("Average hashing/signing time: {:?}", avg_time);
    println!("Total time for {} iterations: {:?}", total_iterations, total_time);
}
